### ISSUE TITLE
Linting with pylint3 sometimes fails

### ISSUE THREAD

**Issue by itaygr** [DonJayamanne/pythonVSCode#807]
## Environment data
VS Code version:  1.11.0-insider (2017-03-03T07:08:55.738Z)
Python Extension version:  0.5.9
Python Version:   Python 3.5.2
OS and version:  Ubuntu 16.04.2 LTS

## Actual behavior

Linting would occasionally fail. 

I have the tool configured to lint whenever I save the file. I believe the issue manifests when I save the file multiple times without giving the linter time to complete after each save.

The message is a slide down error: `There was an error in running the linter 'pylint'`.

## Expected  behavior

Success each time.

## Steps to reproduce:  
-  Checkout a git repo containing a python file
-  Repeat in rapid succession: Make minor modification to the file, Save the file (to invoke linter)
-  See the error message produced

## Logs
Output from ```Python``` output panel  

I've seen multiple different outputs that might give a clue as to what is failing.
1. Incorrect module name 'stock.py.git', should be 'stock.py':
```
##########Linting Output - pylint##########
************* Module /scratch/itay/workspace/mdmaker/src/stock.py.git
1,0,fatal,F0001:No module py.git in src/stock
```

2. Reports an error, even though the linting seems to have worked:
```
Linting with pylint failed.
Command failed: pylint3 --rcfile=/home/itay/.pylint --msg-template='{line},{column},{category},{msg_id}:{msg}' --reports=n --output-format=text /scratch/itay/workspace/mdmaker/src/stock.py

##########Linting Output - pylint##########
************* Module stock
218,0,convention,C0303:Trailing whitespace
39,0,convention,C0103:Invalid constant name "OrderBookLevel"
41,0,convention,C0103:Invalid constant name "Order"
42,0,convention,C0103:Invalid constant name "Trade"
586,31,warning,W0612:Unused variable 'mid_series'
586,43,warning,W0612:Unused variable 'book_mid_series'
19,0,convention,C0411:external import "from recordclass import recordclass" comes before "import numpy"
20,0,convention,C0411:external import "from sortedcontainers import SortedListWithKey" comes before "import numpy"
```

3. Success:

```
##########Linting Output - pylint##########
************* Module stock
39,0,convention,C0103:Invalid constant name "OrderBookLevel"
41,0,convention,C0103:Invalid constant name "Order"
42,0,convention,C0103:Invalid constant name "Trade"
586,43,warning,W0612:Unused variable 'book_mid_series'
586,31,warning,W0612:Unused variable 'mid_series'
19,0,convention,C0411:external import "from recordclass import recordclass" comes before "import numpy"
20,0,convention,C0411:external import "from sortedcontainers import SortedListWithKey" comes before "import numpy"
```
I'd like to also point out the the pylint configuration file is the default configuration with no modifications, as generated by `pylint3 --generate-rcfile`.

**Comment by asterius1** [comment]
Got same issue (although I only seen scenarios 2. and 3.)

VS Code version: 1.10.2
Python Extension version: 0.6.0
Python Version: Python 3.5.2 (Anaconda)
OS and version: Windows 10

Pylint version: 1.6.5

**Comment by pa-eidam** [comment]
Observing the same issue (scenarios 2 and 3).
I'm using pep8, pylint and mypy. Pylint and mypy occasionally fail while saving.

VS Code version: 1.10.2
Python Extension version: 0.6.0
Python Version: Python 3.4.4 (WinPython 32bit)
OS and version: Windows 7 32bit

Pylint version: 1.6.4
Mypy version: 0.470

**Comment by shaypal5** [comment]
Observing the same issue, manifesting as scenarios 2 and 3 (as far as I've seen).
I'm using pylint, which will occasionally fail while saving.

VS Code version: 1.10.2
Python Extension version: 0.6.0
Python Version: Python 3.5.3 (as installed by `pyenv`)
OS and version: OS X El Capitan 10.11.6

Pylint version: 1.6.5

**Comment by DavidNavalho** [comment]
I'm observing the same issue (scenarios 2 and 3), with pylon occasionally failing.

VS Code version: 1.10.2
Python Extension version: 0.6.0
Python Version: Python 3.5.2 (system default)
OS and version: macOS Sierra 10.12.3

Pylint version: 1.6.5

**Comment by DonJayamanne** [comment]
Please reopen the issue if the problem persists.

**Comment by xgid** [comment]
Hi there,

I have just installed VS Code and the [ms-python.python extension](https://github.com/microsoft/vscode-python) and have (randomly) the exact same scenario 1 reported above. ¿Should I re-open this issue here or open a new one at https://github.com/microsoft/vscode-python?

VS Code version: 1.18.1
Python Extension version: 0.9.1
Python Version: Python 3.6.2 (Anaconda 32-bit distribution, system default)
OS and version: Windows 10 64-bits
Pylint version: 1.6.4

**Comment by xgid** [comment]
I've just found that Pylint version 1.6.4 shipped with Anaconda 4.4.0 (which is Python 3.6 based) is in fact incompatible with Python 3.6! See: https://github.com/ContinuumIO/anaconda-issues/issues/2451

That may be the cause of the problem, although I cannot confirm because the problem above happens randomly.
